tests/nightly/test_large_array.py

```diff
@@ -1351,6 +1407,48 @@ def check_tan():
         expected_output = [-.577, -1, 0, 1, .577]
         assert_correctness_of_trigonometric_ops(y, expected_output)
 
+    def check_arcsinh():
+        x = create_input_for_trigonometric_ops([-np.pi/2, -np.pi/4, 0, np.pi/4, np.pi/2])
+        y = nd.arcsinh(x)
+        # expected ouput for indices=(0, 1, -3, -2, -1) after applying arcsinh()
+        expected_output = [np.arcsinh(-np.pi/2), np.arcsinh(-np.pi/4), 0, np.arcsinh(np.pi/4), np.arcsinh(np.pi/2)]
+        assert_correctness_of_trigonometric_ops(y, expected_output)
+
+    def check_arccosh():
+        x = create_input_for_trigonometric_ops([1, np.pi/2, 3*np.pi/4, np.pi])
+        y = nd.arccosh(x)
+        # expected ouput for indices=(0, 1, -3, -2, -1) after applying arccosh()
+        expected_output = [0, np.arccosh(np.pi/2), np.arccosh(3*np.pi/4), np.arccosh(np.pi)]
+        assert_correctness_of_trigonometric_ops(y, expected_output)
+
+    def check_arctanh():
+        x = create_input_for_trigonometric_ops([-1/4, -1/2, 0, 1/4, 1/2])
+        y = nd.arctanh(x)
+        # expected ouput for indices=(0, 1, -3, -2, -1) after applying arctanh()
+        expected_output = [np.arctanh(-1/4), np.arctanh(-1/2), 0, np.arctanh(1/4), np.arctanh(1/2)]
+        assert_correctness_of_trigonometric_ops(y, expected_output)
+
+    def check_sinh():
+        x = create_input_for_trigonometric_ops([-np.pi/2, -np.pi/4, 0, np.pi/4, np.pi/2])
+        y = nd.sinh(x)
+        # expected ouput for indices=(0, 1, -3, -2, -1) after applying sinh()
+        expected_output = [np.sinh(-np.pi/2), np.sinh(-np.pi/4), 0, np.sinh(np.pi/4), np.sinh(np.pi/2)]
+        assert_correctness_of_trigonometric_ops(y, expected_output)
+
+    def check_cosh():
+        x = create_input_for_trigonometric_ops([0, 1, np.pi/2, 3*np.pi/4, np.pi])
+        y = nd.cosh(x)
+        # expected ouput for indices=(0, 1, -3, -2, -1) after applying cosh()
+        expected_output = [1, np.cosh(1), np.cosh(np.pi/2), np.cosh(3*np.pi/4), np.cosh(np.pi)]
+        assert_correctness_of_trigonometric_ops(y, expected_output)
+
+    def check_tanh():
+        x = create_input_for_trigonometric_ops([-1/4, -1/2, 0, 1/4, 1/2])
+        y = nd.tanh(x)
+        # expected ouput for indices=(0, 1, -3, -2, -1) after applying tanh()
+        expected_output = [np.tanh(-1/4), np.tanh(-1/2), 0, np.tanh(1/4), np.tanh(1/2)]
+        assert_correctness_of_trigonometric_ops(y, expected_output)
```

Comment:
why are trigonometric ops being compared with the `assert_correctness_of_trigonometric_ops` function. why not use `assert_almost_equal` from the test `utils.py` file?

Reply:
@anirudhacharya assert almost equals require ndarray inputs. In Large Tensor tests we are only inspecting certain specific indices of the output. Therefore instead of using `assert_almost_equals` I have written a function that does the follwoing:
`np.abs(output-expected)<=atol`

Reply:
> assert almost equals require ndarray inputs.

`assert_almost_equal` accepts both `np.ndarray` and `mx.nd.array` - [see here](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/test_utils.py#L534-L541)

> In Large Tensor tests we are only inspecting certain specific indices of the output

why certain specific indices only? even if we do not want to compare the whole array, shouldn't the indices at least be randomized?

> that does the follwoing:
> `np.abs(output-expected)<=atol`

it is useful to have both absolute and relative tolerance when comparing arrays with floating point values, because tolerance should be seen as a percentage of the original value. 
```
abs(Expected_Value - Computed_Value) < RTOL * abs(Expected_Value) + ATOL
```
eg. a change of 0.1 does not seem much when comparing 100 and 100.1. But when comparing 0.1 and 0.2 a difference of 0.1 makes a lot of difference and should ideally fail the test.

Reply:
So basically this function is specific to testing only Large Tensors. I agree that `assert_almost_equals` is the ideal way to but check each individual value will consume a lot of memory and will take a lot of time resulting in nightly timeout. This functions checks for certain predetermined indices of the output so the correctness is ensured. Since the initial input is small and then it is made repetitive so this check works. These values have been manually checked by comparing output from ndarray and numpy trigonometric ops and the atol was found to be 1e-3. Thats why only atol is used. Currently `assert_correctness_of_trigonometric_ops` is only specific to use case of large tensors

Reply:
I understand you checking individual values instead of whole arrays for large tensors. but a few concerns in this file - 
 - there seems to be erroneous comments like here - https://github.com/apache/incubator-mxnet/pull/16416/files#diff-9ee911616af04047075035c95cf542fbL1298 in the `create_input_for_trigonometric_ops` function in line 1298. Please clean those up.
 - instead of using hard coded values like 1e-3 in `assert_correctness_of_trigonometric_ops` can they be passed as `atol` arguments to these utility functions. It can have a default value of `1e-3`.

I will merge this now, since these changes are not specific to this PR. But make sure you address these errors and clean up the `test_large_array` file in a subsequent PR.

Reply:
Addressed it here #16585